onnxruntime/core/framework/allocator.cc

```diff
@@ -52,7 +52,13 @@ std::ostream& operator<<(std::ostream& out, const OrtAllocatorInfo& info) {
 
 ORT_API_STATUS_IMPL(OrtCreateAllocatorInfo, _In_ const char* name1, OrtAllocatorType type, int id1,
                     OrtMemType mem_type1, _Out_ OrtAllocatorInfo** out) {
-  *out = new OrtAllocatorInfo(name1, type, id1, mem_type1);
+  if (strcmp(name1, onnxruntime::CPU) == 0) {
+    *out = new OrtAllocatorInfo(name1, type, OrtDevice(), id1, mem_type1);
+  } else if (strcmp(name1, onnxruntime::CUDA) == 0) {
+    *out = new OrtAllocatorInfo(name1, type, OrtDevice(OrtDevice::GPU, OrtDevice::MemType::Default, static_cast <OrtDevice::DeviceId>(id1)), id1, mem_type1);
+  } else if (strcmp(name1, onnxruntime::CUDA_PINNED) == 0) {
+    *out = new OrtAllocatorInfo(name1, type, OrtDevice(OrtDevice::CPU, OrtDevice::MemType::CUDA_PINNED, static_cast <OrtDevice::DeviceId>(id1)), id1, mem_type1);
+  }
```

Comment:
Should we return some error if name is not expected?

Reply:
Would really prefer to see an error message being returned. nullptr == Status::OK() doesn't it?

---
In reply to: [297956214](https://github.com/microsoft/onnxruntime/pull/1268#discussion_r297956214) [](ancestors = 297956214)